extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -434,9 +434,9 @@ && isNotLocatedByCustomTemplateLocator(locatorPatternsBuildItem.getLocationPatte
                 if (!recordClass.isRecord()) {
                     continue;
                 }
-                Type[] componentTypes = recordClass.recordComponents().stream().map(RecordComponentInfo::type)
-                        .toArray(Type[]::new);
-                MethodInfo canonicalConstructor = recordClass.method(MethodDescriptor.INIT, componentTypes);
+                MethodInfo canonicalConstructor = recordClass.method(MethodDescriptor.INIT,
```

Comment:
I cannot use the new `ClassInfo#canonicalRecordConstructor()` here because we need to get this into 3.8.

Reply:
Too bad :-)